android/src/main/java/com/tailscale/ipn/util/TSLog.kt

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+package com.tailscale.ipn.util
+
+import android.util.Log
+import libtailscale.Libtailscale
+
+object TSLog {
+  var libtailscaleWrapper = LibtailscaleWrapper()
+
+  fun d(tag: String?, message: String) {
+    Log.d(tag, message)
+    libtailscaleWrapper.sendLog(tag, message)
+  }
+
+  fun w(tag: String, message: String) {
+    Log.w(tag, message)
+    libtailscaleWrapper.sendLog(tag, message)
+  }
+
+  // Overloaded function without Throwable because Java does not support default parameters
+  @JvmStatic
+  fun e(tag: String?, message: String) {
+    Log.e(tag, message)
+    libtailscaleWrapper.sendLog(tag, message)
+  }
+
+  fun e(tag: String?, message: String, throwable: Throwable? = null) {
+    if (throwable == null) {
+      Log.e(tag, message)
+      libtailscaleWrapper.sendLog(tag, message)
+    } else {
+      Log.e(tag, message, throwable)
+      libtailscaleWrapper.sendLog(tag, "$message ${throwable?.localizedMessage}")
+    }
+  }
```

Comment:
In swift, we have the notion of debug logs (which only show up when you do local debug builds) and verbose logs, which sit behind a flag you can toggle at build or runtime.

Reply:
oh nice, this is a good idea. I think we can use Proguard to strip out logs for release builds, so maybe we could use that to strip verbose/debug logs? or we can specify in build flavors in our build.gradle. I'll do this in a follow up